File: src/main/java/com/estonianport/agendaza/model/Pago.java
package com.estonianport.agendaza.model;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Pago {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column
	private int pago;
	
	@ManyToOne
	@JoinColumn(name = "medio_de_pago_id")
	private MedioDePago medioDePago;

	@Column
	private LocalDateTime fecha;

	@ManyToOne
	@JoinColumn(name = "evento_id")
	private Evento evento;

	@ManyToOne
	@JoinColumn(name = "usuario_id")
	private Usuario usuario;

}

File: src/main/kotlin/com/estonianport/agendaza/model/Pago.kt
package com.estonianport.agendaza.model

import com.estonianport.agendaza.dto.PagoDTO
import com.estonianport.agendaza.model.enums.Concepto
import com.estonianport.agendaza.model.enums.MedioDePago
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.time.LocalDate
import java.time.LocalDateTime

@Entity
 class Pago(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long,

    @Column
    var monto: Double,

    @Column
    @Enumerated(EnumType.STRING)
    var concepto: Concepto,

    @Column
    @Enumerated(EnumType.STRING)
    var medioDePago: MedioDePago,

    @Column
    var fecha: LocalDateTime,

    @ManyToOne
    @JoinColumn(name = "evento_id")
    var evento: Evento,

    @ManyToOne
    @JoinColumn(name = "encargado_id")
    var encargado: Usuario,

    @Column
    var numeroCuota: String? = null,

    @Column
    var fechaBaja: LocalDate? = null
) {

    fun toDTO(): PagoDTO {
        return PagoDTO(
            id = id,
            monto = monto,
            codigo = evento.codigo,
            medioDePago = medioDePago,
            fechaEvento = evento.inicio,
            nombreEvento = evento.nombre,
            concepto = concepto,
            numeroCuota = numeroCuota,
            empresaId = evento.empresa.id,
            usuarioId = encargado.id,
            fecha = fecha
        )
    }
}